torch_xla/csrc/lowering_context.cpp

```diff
@@ -93,16 +93,23 @@ LoweringContext::LoweringContext(
   }
 }
 
+// import xla::Shape.h to inlcude the following defintion.
+static constexpr int64_t kUnboundedSize = std::numeric_limits<int64_t>::min();
 xla::XlaOp LoweringContext::GetParameter(
-    const std::shared_ptr<torch::lazy::BackendData>& data) {
+    const std::shared_ptr<torch::lazy::BackendData>& data,
+    const std::vector<uint32_t>& dynamic_dims) {
```

Comment:
I am abit confuse for this call, so while we are trying to get the parameter version of the data, we also overwrite the `dynamic_dims` field? I guess the right thing to do is to let `data` to carry that information but we don't fully support that at the moment. If that's the case, maybe rename the second parameter to `overwrite_dynamic_dims`. it also should be a `unordered_set` instead of vector....

Reply:
ok, maybe `overwrite_unbounded_dynamic_dims` is a more accurate name..

Reply:
I think the logic is the `dynamic_dims` only takes effect when the parameter is created. Once it's created, the dynamic_dims cannot be changed anymore (We should assert on that).